File: src/main/java/model/Enemy.java
package model;

import java.io.Serializable;

public class Enemy implements Serializable{
	private String name;
	private int id;
	Status status;
	
	public Enemy() {}
	
	public Enemy(String name, Status status) {
		this.name = name;
		this.status = status;
	}
	
	public Enemy(String name, int id,  Status status) {
		this(name, status);
		this.id = id;
	}
	
	public String getName() {
		return this.name;
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	public int getId() {
		return this.id;
	}
	
	public void setId(int id) {
		this.id = id;
	}
	
	public Status getStatus() {
		return this.status;
	}
	
	public void attack(Enemy enemy, Player player) {
		final int amount = (Integer)enemy.status.getAtk().atk - (Integer)player.status.getDef().def;
		if(amount > 0) {
			final int dmg = (Integer)player.status.getHp().hp - amount;
			Hp damagedHp = new Hp(dmg);
			player.status.setHp(damagedHp);
			System.out.printf("%sの攻撃！！%sに%dのダメージを与えた \n", enemy.getName(), player.getName(), dmg);
		}else {
			System.out.printf("%sの攻撃！！%sにダメージを与えられなかった \n", enemy.getName(), player.getName());
		}
		System.out.printf("%sのHP: %d \n",player.getName(), (Integer)player.status.hp.hp);
	}
	
	public boolean isDead(final Enemy enemy) {
		boolean isCheck = false;
		if((Integer)enemy.status.hp.hp <= 0) {
		isCheck = true;
		}
		return isCheck;
	}
}

File: src/main/webapp/JavaScript/enemy.js
const maintext = document.getElementById("maintext");

class Enemy {
	constructor(name, status) {
		this.name = name;
		this.status = status;
	}
	
	 attack(enemy, player) {
	const amount = enemy.status.atk - player.status.def;
	if (amount > 0) {
		const dmg = player.status.hp - amount;
		const actualDmg = Math.max(dmg, 0);
		player.status.hp = actualDmg;
		maintext.innerText=`${enemyName}の攻撃！！${playerName}に${dmg}のダメージを与えた`;
		player.isDead(player);
	} else {
		maintext.innerText=`${enemyName}の攻撃！！ダメージを与えられなかった`;
	}
	maintext.innerText=`${playerName}のHP: ${player.status.hp}`;
}

	 isDead(enemy) {
		let isCheck = false;
		if (enemy.status.hp <= 0) {
		isCheck = true;
	}
}